### PR TITLE
Add react-docgen-typescript

### PR DESCRIPTION
**Pull request by joshwooding**
Related to #103 #2 #79

This PR adds doc-gen via react-docgen-typescript.

| | Before | After|
| --- | --- | --- |
| Sidebar | localhost_52700__path=_story_example-introduction--page | localhost_55358__path=_story_example-introduction--page
| Docs | image | image

I haven't noticed any significant impact to speed. 🎉 

 It's hooked up to the storybook config so it should act as a drop-in for the webpack functionality for typescript types.

### CODE REVIEW

File: packages/example-react-ts/stories/Button.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { ComponentMeta, ComponentStoryObj } from '@storybook/react';
+import { Button } from './Button';
+
+export default {
+  title: 'Example/Button',
+  component: Button,
+  argTypes: {
+    backgroundColor: { control: 'color' },
```
Comment (IanVS) [inline packages/example-react-ts/stories/Button.stories.tsx:8]: If we add a `.storybook/preview.ts` with the contents below, we can remove this `argTypes`, and the actions addon will wire up to any `onX` methods.

```
export const parameters = {
  actions: { argTypesRegex: '^on[A-Z].*' },
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
};
```

**Review by IanVS — COMMENTED** [review]
This is amazing, great work!  I just had a few small comments, but I'm excited to get this released!

File: packages/example-react-ts/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "react": "17.0.2",
+    "react-dom": "17.0.2"
+  },
+  "devDependencies": {
+    "@storybook/addon-a11y": "^6.4.0",
+    "@storybook/addon-docs": "^6.4.0",
+    "@storybook/addon-essentials": "^6.4.0",
+    "@storybook/react": "^6.4.0",
+    "storybook-builder-vite": "workspace:*",
+    "vite": "2.7.0"
```
[the remark this answers is not in the document]
Comment (joshwooding) [reply packages/example-react-ts/package.json:23]: I was curious about this too

File: packages/example-react-ts/stories/Button.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  },
+} as ComponentMeta<typeof Button>;
+
+export const Primary = {
+  args: {
+    primary: true,
+    label: 'Button',
+  },
+} as ComponentStoryObj<typeof Button>;
```
[the remark this answers is not in the document]
Comment (joshwooding) [reply packages/example-react-ts/stories/Button.stories.tsx:17]: good point 👍🏽

File: packages/example-react-ts/.storybook/main.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+  framework: '@storybook/react',
+  stories: ['../stories/**/*.stories.mdx', '../stories/**/*.stories.@(js|jsx|ts|tsx)'],
+  addons: ['@storybook/addon-a11y', '@storybook/addon-links', '@storybook/addon-essentials'],
+  core: {
+    builder: 'storybook-builder-vite',
+  },
+  features: {
+    storyStoreV7: true,
```
Comment (IanVS) [inline packages/example-react-ts/.storybook/main.ts:9]: I'm not 100% positive we should have this turned on in this story, but I think it's fine to leave in, and we can take it out later if we want.

**Review by IanVS — APPROVED** [review]

**Comment by mogas** [comment]
A bit late to the party, but now that docgen works, is there any easy way to disable it if wanted? Thinking about disabling it for dev and enable it when creating a build/deploy release

**Comment by joshwooding** [comment]
> A bit late to the party, but now that docgen works, is there any easy way to disable it if wanted? Thinking about disabling it for dev and enable it when creating a build/deploy release

@mogas You should be able to turn it off using Storybook's config e.g. https://storybook.js.org/docs/react/configure/typescript#mainjs-configuration and setting `{ reactDocgen: false }`

**Pull request merged**